### PR TITLE
Feature read only organization

### PR DESCRIPTION
**Pull request by fedescarpa**
## :dart: Goal
Read only mode

## :memo: Details
* Changes in Application Controller to add minimum role for evry controller (default is student)
* Change minimum role to ex student into profile, FAQs, Book, Chapters, Lessons and Exercises controller and validate user permissions depending of user access mode
* Hiding some view elements if needed

## :warning: Dependencies
* https://github.com/mumuki/mumuki-domain/pull/225
* https://github.com/mumuki/mumukit-auth/pull/45

## :back: Backwards compatibility
Should be 100%

### CODE REVIEW

File: spec/features/menu_bar_spec.rb [hunk 1]
```diff
@@ -50,11 +50,11 @@
     let(:admin) { create(:user, permissions: {student: 'private/*', admin: 'private/*'}) }
     let(:owner) { create(:user, permissions: {student: 'private/*', owner: 'private/*'}) }
 
-    scenario 'visitor should only see their account' do
+    scenario 'visitor should only see sing out' do
```
Comment (flbulgarelli) [inline spec/features/menu_bar_spec.rb:53]: Could we add some more feature tests for the several scenarios of access modes and organization's statuses, please?

**Review by flbulgarelli — CHANGES_REQUESTED** [review]

File: app/controllers/application_controller.rb [hunk 2]
```diff
@@ -24,7 +24,6 @@ class ApplicationController < ActionController::Base
   before_action :redirect_to_proper_context!, if: :immersive_context_wrong?
 
   before_action :authorize_if_private!
-  before_action :validate_active_organization!
```
Comment (fedescarpa) [inline app/controllers/application_controller.rb:27]: I didn't delete this action, it's duplicated, that's why I remove one of both... See line 22 of this file

File: app/helpers/menu_bar_helper.rb [hunk 3]
```diff
@@ -1,12 +1,11 @@
 module MenuBarHelper
   def menu_bar_links
     [
-      menu_link_to_profile,
       menu_link_to_classroom,
       menu_link_to_bibliotheca,
       solve_discussions_link,
       user_discussions_link
-    ]
+    ].compact
```
Comment (julian-berbel) [inline app/helpers/menu_bar_helper.rb:8]: is this needed? :thinking:

File: app/views/book/show.html.erb [hunk 4]
```diff
@@ -6,16 +6,25 @@
...
-    </div>
+    <% if show_content_element? %>
+      <div class="actions">
+        <a href="<%=   lesson_path(lesson)   %>" class="btn btn-complementary">
+          <%=   t book_practice_key_for(current_user)   %>
+        </a>
+      </div>
+    <%  else  %>
+      <br>
+    <%   end %>
```
Comment (julian-berbel) [inline app/views/book/show.html.erb:17]: just being nitpicky but all the double/triple spaces in this file are triggering me

**Review by julian-berbel — APPROVED** [review]
LGTM

**Review by flbulgarelli — APPROVED** [review]
Looks lovely to me :heart:

**Pull request merged**